server/app/tasks/DatabaseSeedTask.java

```diff
@@ -0,0 +1,162 @@
+package tasks;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.ebean.DB;
+import io.ebean.Database;
+import io.ebean.SerializableConflictException;
+import io.ebean.Transaction;
+import io.ebean.TxScope;
+import io.ebean.annotation.TxIsolation;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.NonUniqueResultException;
+import javax.persistence.RollbackException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.i18n.Lang;
+import repository.VersionRepository;
+import services.CiviFormError;
+import services.ErrorAnd;
+import services.LocalizedStrings;
+import services.question.QuestionService;
+import services.question.types.DateQuestionDefinition;
+import services.question.types.NameQuestionDefinition;
+import services.question.types.QuestionDefinition;
+
+/**
+ * Task for seeding the database. All of its work is done in the constructor, which is its only
+ * public method.
```

Comment:
Why does the work need to be done in the constructor? Is that a constraint imposed by the fact that it's a "Task" or is it something different?

Reply:
It's a constraint imposed by Play's method of running code at server start time, which is to bind it as an eager singleton in DI: https://www.playframework.com/documentation/2.8.x/ScheduledTasks#Starting-tasks-when-your-app-starts